src/templates/codebuild-deployment-template.yaml

```diff
@@ -0,0 +1,655 @@
+    Default: "1.0.0"
+
+  pUserDefinedTagKey3:
+    Type: String
+    Description: Key for user defined tag 3
+    AllowedPattern: "^(?!aws:)[A-Za-z0-9\\s_.:=+-@/]{0,128}"
+    ConstraintDescription: "Must match the allowable values for a Tag Key. This can only contain alphanumeric characters or special characters ( _ . : / = + - or @) up to 128 characters"
+
+  pUserDefinedTagValue3:
+    Type: String
+    Description: Value for user defined tag 3
+    AllowedPattern: "^[\\w\\s_.:=+-@/]{0,256}$"
+    ConstraintDescription: "Must match the allowable values for a Tag Value. This can only contain alphanumeric characters or special characters ( _ . : / = + - or @) up to 256 characters"
+
+  pUserDefinedTagKey4:
+    Type: String
+    Description: Key for user defined tag 4
+    AllowedPattern: "^(?!aws:)[A-Za-z0-9\\s_.:=+-@/]{0,128}"
+    ConstraintDescription: "Must match the allowable values for a Tag Key. This can only contain alphanumeric characters or special characters ( _ . : / = + - or @) up to 128 characters"
+
+  pUserDefinedTagValue4:
+    Type: String
+    Description: Value for user defined tag 4
+    AllowedPattern: "^[\\w\\s_.:=+-@/]{0,256}$"
+    ConstraintDescription: "Must match the allowable values for a Tag Value. This can only contain alphanumeric characters or special characters ( _ . : / = + - or @) up to 256 characters"
+
+  pUserDefinedTagKey5:
+    Type: String
+    Description: Key for user defined tag 5
+    AllowedPattern: "^(?!aws:)[A-Za-z0-9\\s_.:=+-@/]{0,128}"
+    ConstraintDescription: "Must match the allowable values for a Tag Key. This can only contain alphanumeric characters or special characters ( _ . : / = + - or @) up to 128 characters"
+
+  pUserDefinedTagValue5:
+    Type: String
+    Description: Value for user defined tag 5
+    AllowedPattern: "^[\\w\\s_.:=+-@/]{0,256}$"
+    ConstraintDescription: "Must match the allowable values for a Tag Value. This can only contain alphanumeric characters or special characters ( _ . : / = + - or @) up to 256 characters"
+
+Conditions:
+  # Conditions to control creation of user-defined tags
+  cCreateTag1: !And
+    - !Not [!Equals [!Ref pUserDefinedTagKey1, '']]
+    - !Not [!Equals [!Ref pUserDefinedTagValue1, '']]
+
+  cCreateTag2: !And
+    - !Not [!Equals [!Ref pUserDefinedTagKey2, '']]
+    - !Not [!Equals [!Ref pUserDefinedTagValue2, '']]
+
+  cCreateTag3: !And
+    - !Not [!Equals [!Ref pUserDefinedTagKey3, '']]
+    - !Not [!Equals [!Ref pUserDefinedTagValue3, '']]
+
+  cCreateTag4: !And
+    - !Not [!Equals [!Ref pUserDefinedTagKey4, '']]
+    - !Not [!Equals [!Ref pUserDefinedTagValue4, '']]
+
+  cCreateTag5: !And
+    - !Not [!Equals [!Ref pUserDefinedTagKey5, '']]
+    - !Not [!Equals [!Ref pUserDefinedTagValue5, '']]
+
+Resources:
+  # Create S3 bucket to store configuration files
+  rConfigBucket:
+    Type: AWS::S3::Bucket
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We recommend that you use S3 Lifecycle configuration whenever defining an S3 bucket in CloudFormation or CDK. Setting lifecycle rules for the bucket helps in cleaning up unused data from buckets which in turn leads to cost and storage savings.

[Learn more](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig.html)